src/AlertsManagement/AlertsManagement/AlertCommands/MeasureAzureAlertStatistic.cs

```diff
@@ -0,0 +1,149 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.AlertsManagement.OutputModels;
+using Microsoft.Azure.Management.AlertsManagement.Models;
+
+namespace Microsoft.Azure.Commands.AlertsManagement
+{
+    [Cmdlet("Measure", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AlertStatistic")]
+    [OutputType(typeof(PSAlertsSummary))]
+    public class MeasureAzureAlertStatistic : AlertsManagementBaseCmdlet
+    {
+        #region Parameters declarations
+
+        /// <summary>
+        /// Group by mentioned property of alert
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   HelpMessage = "Summarize by property")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("severity", "alertState", "monitorCondition", "monitorService", "signalType", "alertRule")]
+        public string GroupBy { get; set; }
+
+        /// <summary>
+        /// Resource Id
+        /// </summary>
+        [Parameter(Mandatory = false,
+                   HelpMessage = "Filter on Resource Id of the target resource of alert.")]
+        public string TargetResource { get; set; }
+
+        /// <summary>
+        /// Resource Type
+        /// </summary>
+        [Parameter(Mandatory = false,
```

Comment:
I think this cmdlet needs some parameter sets.  If it doesn't make sense to filter by both TargetResourceId and TargetResourceType, then these should be in separate parameter sets, and the same goes for other mutually exclusive filtering options

Reply:
Our API is also exposed this way. It would not give any results if mutually exclusive values are provided? Do we need to handle this separately here?

Reply:
We need to create parameter sets for logically mutually exclusive parameters, even if this is not represented in the API.